staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -171,17 +175,21 @@ func WithMaxInFlightLimit(
 				// request is being served and after it is done being
 				// served, because both states contribute to the
 				// sampled stats on concurrency.
-				if isMutatingRequest {
-					watermark.recordMutating(len(c))
-				} else {
-					watermark.recordReadOnly(len(c))
+				if watermarkEnabled {
```

Comment:
Unless I'm missing what that `channel` is doing, this might be a little cleaner to have this under a single `if` block instead of the check again in the defer function, what do you think?